integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/iceberg/SparkIcebergCatalogIT.java

```diff
@@ -239,6 +243,75 @@ void testIcebergTableRowLevelOperations(IcebergTableWriteProperties icebergTable
     testIcebergMergeIntoUpdateOperation(icebergTableWriteProperties);
   }
 
+  @Test
+  void testIcebergAsOfQuery() throws NoSuchTableException {
+    String tableName = "test_iceberg_as_of_query";
+    dropTableIfExists(tableName);
+    createSimpleTable(tableName);
+    checkTableColumns(tableName, getSimpleTableColumn(), getTableInfo(tableName));
+
+    sql(String.format("INSERT INTO %s VALUES (1, '1', 1)", tableName));
+    List<String> tableData = getQueryData(getSelectAllSqlWithOrder(tableName, "id"));
+    Assertions.assertEquals(1, tableData.size());
+    Assertions.assertEquals("1,1,1", tableData.get(0));
+
+    SparkIcebergTable sparkIcebergTable = getSparkIcebergTableInstance(tableName);
+    long snapshotId = getCurrentSnapshotId(tableName);
+    sparkIcebergTable.table().manageSnapshots().createBranch("test_branch", snapshotId).commit();
+    sparkIcebergTable.table().manageSnapshots().createTag("test_tag", snapshotId).commit();
+    long snapshotTimestamp = getCurrentSnapshotTimestamp(tableName);
+    long timestamp = waitUntilAfter(snapshotTimestamp + 1000);
+    waitUntilAfter(timestamp + 1000);
```

Comment:
`waitUntilAfter(timestamp + 1000);`  should be removed?

Reply:
ok. done.